activerecord/lib/active_record/relation/query_methods.rb

```diff
@@ -1312,6 +1312,10 @@ def distinct(value = true)
 
     # Like #distinct, but modifies relation in place.
     def distinct!(value = true) # :nodoc:
+      unless [true, false].include?(value)
+        raise ArgumentError, "Expected TrueClass or FalseClass, got #{value.class}"
+      end
```

Comment:
```suggestion
      unless value == true || value == false
        raise ArgumentError, "distinct expects a boolean value, got: #{value.inspect}"
      end
```

Reply:
This logic should be placed in `distinct` method, not `distinct!`.

Reply:
Should we not keep it consistent acrross both even if the latter is undocumented?

Reply:
Ok, I was sure that ActiveRecord prefers the suggested style inside that file, but I see examples of both error checking in user-facing and internal `!` methods.
So using it inside `distinct!` seems ok.

Reply:
I have committed your suggestion but kept the validation in both `#distinct` and `distinct!`.